### PR TITLE
Dp subset tests Closes #9943

### PR DESCRIPTION
**Pull request by aryandgandhi**
### Describe your change:

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by aryandgandhi** [comment]
@cclauss @tianyizheng02 having trouble adding type annotations, there is an issue with ruff and mypy expectecing different annotations
Screenshot 2023-10-14 at 4 19 54 PM
Screenshot 2023-10-14 at 4 36 14 PM

**Comment by aryandgandhi** [comment]
Added docstring tests, add hacktoberfest tag when possible

**Comment by aryandgandhi** [comment]
CC. @dhruvmanila

File: dynamic_programming/subset_generation.py [hunk 1]
```diff
@@ -9,16 +9,32 @@ def combination_util(arr, n, r, index, data, i):
...
+
+    >>> combination_util([1], 1, 2, 0, [0, 0], 3)
+
+
     """
     if index == r:
-        for j in range(r):
-            print(data[j], end=" ")
-        print(" ")
+        print(" ".join(map(str, data[:r])))
```
Comment (cclauss) [inline dynamic_programming/subset_generation.py:31]: CONTRIBUTING.md says that algorithmic functions should not `print()` instead they should return or yield values or raise exceptions.

**Comment by aryandgandhi** [comment]
updated the print @cclauss

**Comment by cclauss** [comment]
algorithmic functions should not print()

**Comment by aryandgandhi** [comment]
> algorithmic functions should not print()

Even in the main?

**Comment by cclauss** [comment]
`main()` can print because it is a utility function, not an algorithmic function.

**Comment by aryandgandhi** [comment]
> `main()` can print because it is a utility function, not an algorithmic function.

Yeah I removed the print from algorithmic function

**Comment by aryandgandhi** [comment]
@cclauss can this be merged with Hacktoberfest commit, or any changes you still reocmmend

**Comment by cclauss** [comment]
# This branch has conflicts that must be resolved

See below.

**Comment by cclauss** [comment]
Closing in favor of:
* #10191 (merged)

**Pull request closed without merging**